src/Microsoft.OpenApi.Readers/OpenApiReaderSettingsExtensions.cs

```diff
@@ -0,0 +1,33 @@
+using Microsoft.OpenApi.Interfaces;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Readers;
+
+namespace Microsoft.OpenApi.Reader;
+
+/// <summary>
+/// Extensions for <see cref="OpenApiReaderSettings"/>
+/// </summary>
+public static class OpenApiReaderSettingsExtensions
+{
+    /// <summary>
+    /// Adds a reader for the specified format
+    /// </summary>
+    /// <param name="settings">The settings to add the reader to.</param>
+    public static void AddYamlReader(this OpenApiReaderSettings settings)
+    {
+        var yamlReader = new OpenApiYamlReader();
+        settings.AddReaderToSettings(OpenApiConstants.Yaml, yamlReader);
+        settings.AddReaderToSettings(OpenApiConstants.Yml, yamlReader);
+    }
+    private static void AddReaderToSettings(this OpenApiReaderSettings settings, string format, IOpenApiReader reader)
+    {
+#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP || NET5_0_OR_GREATER
+        settings.Readers.Add(format, reader);
```

Comment:
Should we use `TryAdd` instead or are we keen on an exception being thrown if the key already exists in the dictionary?

Reply:
Sorry. Ignore my previous comment. Looks like it will still throw in latest versions of NET according to the docs. I was under the impression the behaviour changed. 
https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.add?view=net-9.0

Reply:
Thanks, that was an oversight, I meant to use try add here hence the conditional compilation